src/document.c

```diff
@@ -930,36 +930,89 @@ typedef struct
 } FileData;
 
 
+static time_t get_mtime(const gchar *locale_filename)
+{
+	GError *error = NULL;
+	const gchar *err_msg = NULL;
+	time_t time = 0;
+
+	if (!file_prefs.use_safe_file_saving && file_prefs.use_gio_unsafe_file_saving)
+	{
+		GFile *file = g_file_new_for_path(locale_filename);
+		GFileInfo *info = g_file_query_info(file, G_FILE_ATTRIBUTE_TIME_MODIFIED, G_FILE_QUERY_INFO_NONE, NULL, &error);
+
+		if (info)
+		{
+			GTimeVal timeval;
+
+			g_file_info_get_modification_time(info, &timeval);
+			g_object_unref(info);
+			time = timeval.tv_sec;
+		}
+		else if (error)
+			err_msg = error->message;
+
+		g_object_unref(file);
+	}
+	else
+	{
+		struct stat st;
```

Comment:
`g_stat()` uses different sized `stat()` calls on different platforms (win32/64 etc) so this should be `GStatBuf` that matches whatever `g_stat()` does.